cmd/antrea-agent/agent.go

```diff
@@ -140,6 +140,7 @@ func run(o *Options) error {
 	enableMulticlusterGW := features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.EnableGateway
 	enableMulticlusterNP := features.DefaultFeatureGate.Enabled(features.Multicluster) && o.config.Multicluster.EnableStretchedNetworkPolicy
 	enableFLowExporter := features.DefaultFeatureGate.Enabled(features.FlowExporter) && o.config.FlowExporter.Enable
+	enableAntreaProxy := *o.config.AntreaProxy.Enable
```

Comment:
I think it should check both feature gate and the flag like the above option, otherwise users who disable featuregate will see unexpected behavior after upgrade. We should add a logging when users disable the feature gate but enables the flag to indicate that the feature has graduated to GA, they should use the flag to disable it in the future.

Reply:
Agreed, I added some code for compatibility with AntreaProxy disabled:

```
	enableAntreaProxy := *o.config.AntreaProxy.Enable
	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
		enableAntreaProxy = false
		klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
	}
```